src/irmin/commit.ml

```diff
@@ -558,35 +558,24 @@ module V1 = struct
 
   module Make (C : Generic_key.S with module Info := Info) = struct
     module K (K : Type.S) = struct
-      let h = Type.string_of `Int64
-      let hash_to_bin_string = Type.(unstage (to_bin_string K.t))
-      let hash_of_bin_string = Type.(unstage (of_bin_string K.t))
-      let size_of = Type.Size.using hash_to_bin_string (Type.Size.t h)
-
-      let encode_bin =
-        let encode_bin = Type.(unstage (encode_bin h)) in
-        fun e k -> encode_bin (hash_to_bin_string e) k
-
-      let decode_bin =
-        let decode_bin = Type.(unstage (decode_bin h)) in
-        fun buf pos_ref ->
-          let v = decode_bin buf pos_ref in
-          match hash_of_bin_string v with
-          | Ok v -> v
-          | Error (`Msg e) -> Fmt.failwith "decode_bin: %s" e
-
       type t = K.t
 
       let pre_hash = Type.(unstage (pre_hash K.t))
+      let size = Type.(unstage (size_of K.t))
 
-      let pre_hash t f =
-        let f x =
-          f "\000\000\000\000\000\000\000 ";
-          f x
+      let pre_hash x f =
+        let () =
+          match size x with
+          | None -> ()
+          | Some i ->
+              let b = Bytes.create 8 in
+              Bytes.set_int64_be b 0 (Int64.of_int i);
+              let b = Bytes.unsafe_to_string b in
```

Comment:
The size of a hash is fixed. Let's compute that string only once at the module level

Reply:
Fixed, thanks for your help.